src/Umbraco.PublishedCache.NuCache/Property.cs

```diff
@@ -33,7 +33,7 @@ internal class Property : PublishedPropertyBase
     private object? _interValue;
 
     // the variant source and inter values
-    private Dictionary<CompositeStringStringKey, SourceInterValue>? _sourceValues;
+    private readonly Lazy<ConcurrentDictionary<CompositeStringStringKey, SourceInterValue>> _sourceValues = new();
```

Comment:
Just create the dictionary and avoid Lazy completely?

Reply:
Same as above, if creating every time is more efficient overall then happy to go with that - but keen to make sure.

Reply:
Agree. Lazy has historically caused a lot of trouble with disposal and memory leakage.

Reply:
Hmm. Come to think of it, the `_sourceValues` dictionary is only created for variant properties. In a LOT of cases, the property is not going to be variant, so the up-front allocation of a dictionary is an overhead compared to today.

Perhaps the `Lazy` approach is justified here.

Reply:
Just browsing a bit through the code it seems this _sourceValues is used quite a lot (either through HasValue or GetSourceValue).